src/OrleansAWSUtils/Membership/SiloInstanceRecord.cs

```diff
@@ -0,0 +1,210 @@
+        public const string PROXY_PORT_PROPERTY_NAME = "ProxyPort";
+        public const string SILO_NAME_PROPERTY_NAME = "SiloName";
+        public const string INSTANCE_NAME_PROPERTY_NAME = "InstanceName";
+        public const string SUSPECTING_SILOS_PROPERTY_NAME = "SuspectingSilos";
+        public const string SUSPECTING_TIMES_PROPERTY_NAME = "SuspectingTimes";
+        public const string START_TIME_PROPERTY_NAME = "StartTime";
+        public const string I_AM_ALIVE_TIME_PROPERTY_NAME = "IAmAliveTime";
+        internal const char Seperator = '-';
+
+        public SiloInstanceRecord() { }
+
+        public SiloInstanceRecord(Dictionary<string, AttributeValue> fields)
+        {
+            if (fields.ContainsKey(DEPLOYMENT_ID_PROPERTY_NAME))
+                DeploymentId = fields[DEPLOYMENT_ID_PROPERTY_NAME].S;
+
+            if (fields.ContainsKey(SILO_IDENTITY_PROPERTY_NAME))
+                SiloIdentity = fields[SILO_IDENTITY_PROPERTY_NAME].S;
+
+            if (fields.ContainsKey(ADDRESS_PROPERTY_NAME))
+                Address = fields[ADDRESS_PROPERTY_NAME].S;
+
+            int port;
+            if (fields.ContainsKey(PORT_PROPERTY_NAME) && 
+                int.TryParse(fields[PORT_PROPERTY_NAME].N, out port))
+                Port = port;
+
+            int generation;
+            if (fields.ContainsKey(GENERATION_PROPERTY_NAME) &&
+                int.TryParse(fields[GENERATION_PROPERTY_NAME].N, out generation))
+                Generation = generation;
+
+            if (fields.ContainsKey(HOSTNAME_PROPERTY_NAME))
+                HostName = fields[HOSTNAME_PROPERTY_NAME].S;
+
+            int status;
+            if (fields.ContainsKey(STATUS_PROPERTY_NAME) &&
+                int.TryParse(fields[STATUS_PROPERTY_NAME].N, out status))
+                Status = status;
+
+            int proxyPort;
+            if (fields.ContainsKey(PROXY_PORT_PROPERTY_NAME) &&
+                int.TryParse(fields[PROXY_PORT_PROPERTY_NAME].N, out proxyPort))
+                ProxyPort = proxyPort;
+
+            if (fields.ContainsKey(SILO_NAME_PROPERTY_NAME))
+                SiloName = fields[SILO_NAME_PROPERTY_NAME].S;
+
+            if (fields.ContainsKey(SUSPECTING_SILOS_PROPERTY_NAME))
+                SuspectingSilos = fields[SUSPECTING_SILOS_PROPERTY_NAME].S;
+
+            if (fields.ContainsKey(SUSPECTING_TIMES_PROPERTY_NAME))
+                SuspectingTimes = fields[SUSPECTING_TIMES_PROPERTY_NAME].S;
+
+            if (fields.ContainsKey(START_TIME_PROPERTY_NAME))
+                StartTime = fields[START_TIME_PROPERTY_NAME].S;
+
+            if (fields.ContainsKey(I_AM_ALIVE_TIME_PROPERTY_NAME))
+                IAmAliveTime = fields[I_AM_ALIVE_TIME_PROPERTY_NAME].S;
+
+            int etag;
+            if (fields.ContainsKey(ETAG_PROPERTY_NAME) &&
+                int.TryParse(fields[ETAG_PROPERTY_NAME].N, out etag))
+                ETag = etag;
+        }
+        
+        public string DeploymentId { get; set; }
```

Comment:
I don't know if there's any specific rule in the project about member variables, but I personally believe not placing them in the beginning of the class declaration and instead in the body makes it harder to grasp the size and shape of the class.